### DISCUSSION TITLE
Usage Instructions (How to Use Guide)

### DISCUSSION THREAD

**Discussion by christiaangoossens** [christiaangoossens/hass-oidc-auth#3] [Announcements]
1. Install the repository from HACS. Setup a Custom Repository with this link: https://github.com/christiaangoossens/hass-oidc-auth and type integration. You should now be able to select the alpha release to download.
2. Add the necessary fields to the configuration, see README.
3. Open a new tab and go to `{your URL}/auth/oidc/welcome`. You should now see:

![image](https://github.com/user-attachments/assets/7320b7d3-b9f9-4268-ba1f-4deb0c6805ea)

If you have configured everything correctly, you should be redirected to your OIDC Provider after clicking the button. Please login there.

4. You should return to a screen like this:

![image](https://github.com/user-attachments/assets/d9c305bd-4a93-4a97-ae55-dba6361d92c8)

Either click the automatic sign in button or copy the code.
This screen will give you a one-time code to login that expires in 5 minutes.

---
**If you copied the code, continue here.**

5. If you go to Home Assistant, you should now see something like this:

![image](https://github.com/user-attachments/assets/4ed2b408-53e4-429e-920a-7628ddbcfc02)

If you don't, you likely see:

![image](https://github.com/user-attachments/assets/80629c60-793e-4933-8b45-283234798ffb)

If so, click "OpenID Connect (SSO)" to get to the first screen.

7. Enter your code into the single input field:

![image](https://github.com/user-attachments/assets/f031a41c-5a85-44b8-8517-3feabaa44fd5)

Upon clicking login, you should now login.
If the code is wrong, you will see this instead:

![image](https://github.com/user-attachments/assets/317d20e4-0e10-40f7-bb68-5cf456faf87d)

**Comment by bobdivx** [comment]
Hello and thank you for your work !
I'm trying to configure SSO with Zitadel, but I have to admit that I'm stuck.

I have the page 
![IMG_20250106_083729](https://github.com/user-attachments/assets/196e207b-23d4-4e38-9845-c910cb6dd657)

However I have a doubt about my zitadel configuration:
![IMG_20250106_084533](https://github.com/user-attachments/assets/1140616e-0597-425e-843b-f11d225d8751)
![Screenshot_2025-01-06-08-41-34-46_40deb401b9ffe8e1df2f1cc5ba480b12](https://github.com/user-attachments/assets/4c25b342-87de-4128-ad61-36ec2189548f)
![Screenshot_2025-01-06-08-39-03-69_40deb401b9ffe8e1df2f1cc5ba480b12](https://github.com/user-attachments/assets/05dd1338-490d-4cb6-a088-c5787501eece)


Thanks in advance

**Reply by christiaangoossens** [reply]
Hi @bobdivx, from your picture I can only see that the callback URL is configured wrong, it should be "/auth/oidc/callback" instead of "/auth/external/callback". Can you let me know which error you get throughout the login process?

**Reply by bobdivx** [reply]
Thank you for taking the time to answer me, so I modified the URL and same result.
Login failed.
Integration is misconfigured, discovery could not be obtained.

Here is my home assistant setup:
auth_oidc:
      client_id: ########
      discovery_url: https://sso.#####/oauth/v2/.well-known/openid-configuration
      features:
        automatic_user_linking: true

**Reply by christiaangoossens** [reply]
Can you check if the discovery url gives you the expected json page?

Also, are you sure that your Home Assistant instance can access that URL too? (Firewall, local DNS)

**Reply by bobdivx** [reply]
Yes the page returns a json. Maybe you can contact me via Messenger to share the personal data with you? That would be easier to resolve this.?
JESER MATHIEU

Home assistant and Zitadel are both deployed in my local Kubernetes cluster and the addresses are well accessible from outside.

Here is the content of the json 
`{
  "issuer": "https://[DOMAINE_MASQUÉ]", 
  "authorization_endpoint": "https://[DOMAINE_MASQUÉ]/oauth/v2/authorize",
  "token_endpoint": "https://[DOMAINE_MASQUÉ]/oauth/v2/token",
  "introspection_endpoint": "https://[DOMAINE_MASQUÉ]/oauth/v2/introspect",
  "userinfo_endpoint": "https://[DOMAINE_MASQUÉ]/oidc/v1/userinfo",
  "revocation_endpoint": "https://[DOMAINE_MASQUÉ]/oauth/v2/revoke",
  "end_session_endpoint": "https://[DOMAINE_MASQUÉ]/oidc/v1/end_session",
  "device_authorization_endpoint": "https://[DOMAINE_MASQUÉ]/oauth/v2/device_authorization",
  "jwks_uri": "https://[DOMAINE_MASQUÉ]/oauth/v2/keys",
  "scopes_supported": ["openid", "profile", "email", "phone", "address", "offline_access"],
  "response_types_supported": ["code", "id_token", "id_token token"],
  "response_modes_supported": ["query", "fragment", "form_post"],
  "grant_types_supported": ["authorization_code", "implicit", "refresh_token", "client_credentials", "urn:ietf:params:oauth:grant-type:jwt-bearer", "urn:ietf:params:oauth:grant-type:device_code"],
  "subject_types_supported": ["public"],
  "id_token_signing_alg_values_supported": ["RS256"],
  "request_object_signing_alg_values_supported": ["RS256"],
  "token_endpoint_auth_methods_supported": ["none", "client_secret_basic", "client_secret_post", "private_key_jwt"],
  "token_endpoint_auth_signing_alg_values_supported": ["RS256"],
  "revocation_endpoint_auth_methods_supported": ["none", "client_secret_basic", "client_secret_post", "private_key_jwt"],
  "revocation_endpoint_auth_signing_alg_values_supported": ["RS256"],
  "introspection_endpoint_auth_methods_supported": ["client_secret_basic", "private_key_jwt"],
  "introspection_endpoint_auth_signing_alg_values_supported": ["RS256"],
  "claims_supported": ["sub", "aud", "exp", "iat", "iss", "auth_time", "nonce", "acr", "amr", "c_hash", "at_hash", "act", "scopes", "client_id", "azp", "preferred_username", "name", "family_name", "given_name", "locale", "email", "email_verified", "phone_number", "phone_number_verified"],
  "code_challenge_methods_supported": ["S256"],
  "ui_locales_supported": ["bg", "cs", "de", "en", "es", "fr", "hu", "id", "it", "ja", "ko", "mk", "nl", "pl", "pt", "ru", "sv", "zh"],
  "request_parameter_supported": true,
  "request_uri_parameter_supported": false
}
`

**Reply by christiaangoossens** [reply]
@bobdivx I have moved the question over to issue #18.

**Reply by christiaangoossens** [reply]
Resolved, for Zitadel, make sure to use `https://{your zitadel url}/.well-known/openid-configuration` as the discovery URL.

**Comment by christiaangoossens** [comment]
Replaced with https://github.com/christiaangoossens/hass-oidc-auth/blob/main/docs/usage.md
